custom_model_runner/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py

```diff
@@ -49,5 +51,9 @@ def _materialize(self, parent_data_objs, user_data):
         input_filename = self._params["input_filename"]
         predictions = self._predictor.predict(input_filename)
         output_filename = self._params.get("output_filename")
-        predictions.to_csv(output_filename, index=False)
-        return [predictions]
+        if self._unstructured_mode:
+            with open(output_filename, "w") as f:
+                f.write(predictions)
+        else:
+            predictions.to_csv(output_filename, index=False)
+        return []
```

Comment:
This used to return predictions in the list, why did this change?

Reply:
This is mlpiper component. It returns a list of objects to pass to the next component in mlpiper pipeline.
Our prediction pipelines contain one component only, so they don't receive anything from the prev component and don't pass anything to the next.
What I forgot - is to fix component.json  file.
Fixed now.